projects/cm_mcu/InitTask.c

```diff
@@ -46,19 +46,17 @@ void InitTask(void *parameters)
   log_info(LOG_SERVICE, "Clock I/O expander initialized\r\n");
 #ifdef REV2
   // grab the semaphore to ensure unique access to I2C controller
-  while (xSemaphoreTake(i2c2_sem, (TickType_t)10) == pdFALSE)
-    ;
+  if (acquireI2CSemaphore(i2c2_sem) == pdFAIL) {
+    log_warn(LOG_SERVICE, "could not get semaphore in time\r\n");
```

Comment:
it looks like we continue here, even if we don't acquire the semaphore in time. Is that the right thing to do?

Reply:
I think it is fine to continue and just give a warning even without a semaphore being taken by these tasks (which run infinitely anyway) because we have the semaphore check of releasing it if acquired at the end. 

From the snapshot in the CLI test above, you can see that the program still runs and it gives us some warnings so a user can just release the corresponding semaphore when they see them :
```
% semaphore 2 take                                                                                                                                                                  
semaphore: taking semaphore 2                                                                                                                                                       
% loadclock 0                                                                                                                                                                       
202211101707 SRV WRN MonitorI2CTask.c:89:could not get semaphore in time                                                                                                            
202211101707 SRV WRN MonitorI2CTask.c:89:could not get semaphore in time                                                                                                            
loadclock is programming clock r0a.                                                                                                                                                 
loadclock: could not get semaphore in time 
```

Perhaps, I could have specifically added in this warning what semaphore device is not available.

Reply:
but if we try to use the i2c device when a different task (which has the semaphore) acts, isn't that a problem?

Reply:
I guess this example illustrate your concern :
```
% semaphore 2 take                                                                                                      
semaphore: taking semaphore 2                                                                                           
% i2cr 2 77 1                                                                                                           
i2cr: dev 2, addr 0x77: val=0x00 00 00 0f                                                                               
% 0202211111357 SRV WRN MonitorI2CTask.c:89:could not get semaphore in time
202211111357 SRV WRN MonitorI2CTask.c:89:could not get semaphore in time
202211111357 MONI2C WRN MonitorI2CTask.c:160:Mux write error ADDR_ACK_ERROR, break (instance=CLKR0A,ps=0)
202211111357 MONI2C ERR MonitorI2CTask.c:184:CLKSI: LOSXAXB read Error ADDR_ACK_ERROR, break (ps=1)
% 202211111358 SRV WRN MonitorI2CTask.c:89:could not get semaphore in time
202211111358 SRV WRN MonitorI2CTask.c:89:could not get semaphore in time
202211111358 MONI2C WRN MonitorI2CTask.c:160:Mux write error ADDR_ACK_ERROR, break (instance=CLKSI,ps=0)
202211111358 MONI2C ERR MonitorI2CTask.c:184:CLKR0A: I2C_ADDR read Error ADDR_ACK_ERROR, break (ps=0)
%  
```

but once I release the semaphore the task that tried to access it will be back running again without errors. I can also try to skip/break the whole action of a task that tries to access a semaphore but I am not really sure how to write a code in order to do so.

Reply:
it looks like in this piece of code, the `InitTask` _still_ continues to run `init_load_clk` without first ensuring it has the semaphore?

Reply:
we resolve this issue by utilizing a new function `acquireI2CSemaphoreBlock` instead so that it will try to grab a semaphore infinitely in the InitTask before starting to program our clocks.